### PR TITLE
CommonServerPython: added the execute_command function

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33327

## Description
added the execute_command function

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py [hunk 1]
```diff
@@ -3323,6 +3323,79 @@ def test_return_results_mixed_results(mocker):
...
+            - Calling execute_command.
+        Then:
+            - Assert that only the Contents value is returned.
+        """
+        from CommonServerPython import execute_command, EntryType
+        demisto_execute_mock = mocker.patch.object(demisto, 'executeCommand',
+                                                   return_value={'Type': EntryType.NOTE,
+                                                                 'Contents': {'hello': 'world'}})
+        res = execute_command('command', {'arg1': 'value'})
+        assert demisto_execute_mock.call_count == 1
```
Comment (Itay4) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:3342]: please also verify it is called with the expected args

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py [hunk 2]
```diff
@@ -3323,6 +3323,79 @@ def test_return_results_mixed_results(mocker):
     assert demisto_results_mock.call_args_list[1][0][0] == mock_demisto_results_entry
 
 
+class TestExecuteCommand():
```
Comment (Itay4) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:3326]: can we also add test for the other cases in the method so we have full coverage (e.g. extract_contents=False, the response from `executeCommand` is not a dict or a list, the response from `executeCommand` is a list with one dict)?

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -5886,6 +5886,42 @@ def return_warning(message, exit=False, warning='', outputs=None, ignore_auto_ex
         sys.exit(0)
 
 
+def execute_command(command, args, extract_contents=True):
+    """
+    Run demisto.executeCommand and check for errors.
```
Comment (Itay4) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:5891]:
```suggestion
    Runs the `demisto.executeCommand()` function and checks for errors.
```

**Review by Itay4 — COMMENTED** [review]
nice work

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -5886,6 +5886,42 @@ def return_warning(message, exit=False, warning='', outputs=None, ignore_auto_ex
...
+
+    :return: The command results
+    :rtype: ``list`` or ``dict`` or ``str``
+    """
+    if not hasattr(demisto, 'executeCommand'):
+        raise DemistoException('Cannot run execute command from integrations.')
+
+    res = demisto.executeCommand(command, args)
+    if is_error(res):
+        return_results(res)
```
[the remark this answers is not in the document]
Comment (yaakovi) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:5910]: waited for the review.
I also considered not returning the full `res` and returning the error using the `get_error` function:
https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L956
worst case, we can log the original entries.

WDYT?

[the remark this answers is not in the document]
Comment (Itay4) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:5910]: yea lets go with `get_error`, i think it is indicative enough

**Comment by yaakovi** [comment]
@roysagi @amshamah419 
When reviewing this PR, make sure to go over the docstring of the function since this is going to be shown in the product as part of the `Script Helper`

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py (hunk 2 above)
Comment (reutshal) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:3326]: @yaakovi Add more tests.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py [hunk 5]
```diff
@@ -3323,6 +3323,79 @@ def test_return_results_mixed_results(mocker):
...
+    @staticmethod
+    def test_sanity(mocker):
+        """
+        Given:
+            - A command with a single entry as output.
```
Comment (reutshal) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:3331]:
```suggestion
            - A successfull command with a single entry as output.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py [hunk 6]
```diff
@@ -3323,6 +3323,79 @@ def test_return_results_mixed_results(mocker):
...
+                                                                 'Contents': {'hello': 'world'}})
+        res = execute_command('command', {'arg1': 'value'})
+        assert demisto_execute_mock.call_count == 1
+        assert res == {'hello': 'world'}
+
+    @staticmethod
+    def test_multiple_results(mocker):
+        """
+        Given:
+            - A command with several entries as output.
```
Comment (reutshal) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:3349]:
```suggestion
            - A successfull command with several entries as output.
```

**Review by reutshal — APPROVED** [review]

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 7]
```diff
@@ -5886,6 +5886,42 @@ def return_warning(message, exit=False, warning='', outputs=None, ignore_auto_ex
...
+    :type command: ``str``
+    :param command: The command to run. (required)
+
+    :type args: ``dict``
+    :param args: The command arguments. (required)
+
+    :type extract_contents: ``bool``
+    :param extract_contents: Whether to return only the Contents part of the results. Default is True.
+
+    :return: The command results
```
Comment (amshamah419) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:5902]:
```suggestion
    :return: The command results.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py (hunk 5 above)
Comment (amshamah419) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:3331]:
```suggestion
            - A successful command with a single entry as output.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py (hunk 6 above)
Comment (amshamah419) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:3349]:
```suggestion
            - A successful command with several entries as output.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 8]
```diff
@@ -5886,6 +5886,42 @@ def return_warning(message, exit=False, warning='', outputs=None, ignore_auto_ex
...
+    res = demisto.executeCommand(command, args)
+    if is_error(res):
+        return_results(res)
+        return_error('Failed to execute {}. See additional error details in the above entries.'.format(command))
+
+    if not extract_contents:
+        return res
+
+    if isinstance(res, dict):
+        return res.get('Contents', {})
```
[the remark this answers is not in the document]
Comment (yaakovi) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:5917]: after some testing, I'm not sure this is possible, I think we will always get a list from the `executeCommand` method

**Pull request merged**